v-next/core/src/internal/plugins/plugin-validation.ts

```diff
@@ -1,6 +1,110 @@
+import { createRequire } from "node:module";
+import path from "node:path";
+import process from "node:process";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import semver from "semver";
+
 import { HardhatPlugin } from "../../types/plugins.js";
 
-export async function validatePluginNpmDependencies(_plugin: HardhatPlugin) {
-  // TODO: If it has an npm package, validate their peer dependencies
-  // If any is missing, throw
+/**
+ * A simplified version of the package.json object
+ */
+interface PackageJson {
+  version: string;
+  peerDependencies: {
+    [name: string]: string;
+  };
+}
+
+/**
+ * Validate that a plugin is installed and that its peer dependencies are installed and satisfy the version constraints.
+ *
+ * @param plugin - the plugin to be validated
+ * @param basePathForNpmResolution - the directory path to use for node module resolution, defaulting to `process.cwd()`
+ */
+export async function validatePluginNpmDependencies(
+  plugin: HardhatPlugin,
+  // TODO: When all loads where based on a hardhat config file
+  // on disk we used the config file's directory as the base path
+  // We need to decide if the current working directory is the right
+  // option for module resolution now
+  basePathForNpmResolution?: string,
```

Comment:
This is for testing currently. Though we need to consider passing in the npm resolution folder when we are loading the config file from disk.